modules/ztp-image-based-upgrade-with-rhacm.adoc

```diff
@@ -0,0 +1,362 @@
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: restore-example-app
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "2"
+spec:
+  backupName:
+    backup-example-app
+----
+
+[NOTE]
+====
+If you do not define the `lca.openshift.io/apply-wave` annotation in the `Backup` or `Restore` CRs, they will be applied together.
+====
+--
+
+. Generate a `ConfigMap` object for your OADP CRs.
+
+.. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-oadp-cm --from-file=example-oadp-resources.yaml=<path_to_oadp_crs> -n openshift-adp
+----
+
+.. Apply the `ConfigMap`:
++
+[source,terminal]
+----
+$ oc apply -f oadp-cm-example.yaml
+----
+
+. (Optional) Create `ConfigMap` resources that contain the additional manifests that you want to apply to the target cluster.
+
+.. Create a YAML file that contains your extra manifestss.
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: "pci-sriov-net-e5l"
+  namespace: openshift-sriov-network-operator
+spec:
+  deviceType: vfio-pci
+  isRdma: false
+  nicSelector:
+    pfNames: [ens1f0]
+  nodeSelector:
+    node-role.kubernetes.io/master: ""
+  mtu: 1500
+  numVfs: 8
+  priority: 99
+  resourceName: pci_sriov_net_e5l
+---
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: "networking-e5l"
+  namespace: openshift-sriov-network-operator
+spec:
+  ipam: |-
+    {
+    }
+  linkState: auto
+  networkNamespace: flexran-testcases
+  resourceName: pci_sriov_net_e5l
+  spoofChk: "on"
+  trust: "off"
+----
+
+.. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-extra-manifests-cm --from-file=example-extra-manifests.yaml=<path_to_extramanifest> -n openshift-lifecycle-agent
+----
+
+. (Optional) To keep your custom catalog sources after the upgrade, generate another `ConfigMap` object for your catalog sources and add them to the `spec.extraManifest` field in the `ImageBasedUpgrade` CR. For more information about catalog sources, see xref:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.15/html-single/operators/index#olm-catalogsource_olm-understanding-olm[Catalog source].
+
+.. Create a YAML file that contains the `CatalogSource` CR.
++
+--
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  annotations:
+    target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+  name: example-catalogsources
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  displayName: disconnected-redhat-operators
+  image: quay.io/example-org/example-catalog:v1
+----
+--
+
+.. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-catalogsources-cm --from-file=example-catalogsources.yaml=<path_to_catalogsource_cr> -n openshift-lifecycle-agent
```

Comment:
Looking at "---from-file=example-catalogsources.yaml"
Do we want to clarify that the name example-catalogsources.yaml is arbitrary, i.e. is it clear enough ?

Reply:
I think that the `example` in the name makes it obvious that this is an example for the process, you don't have to use the same name. This makes it easier to read than using something like `<your_extramanifests_file_name>.yaml`. We do this in our docs in multiple places, [here](https://74025--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/ztp-deploying-far-edge-sites#ztp-creating-the-site-secrets_ztp-deploying-far-edge-sites) and [here](https://74025--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/ztp-preparing-the-hub-cluster#ztp-configuring-the-cluster-for-a-disconnected-environment_ztp-preparing-the-hub-cluster), for example.